src/assets/styles/_base.scss

```diff
@@ -8,8 +8,9 @@ body {
   margin: 0;
   tab-size: 2;
   color: var(--black-500);
-  font-size: 1em;
   font-family: var(--font-family-sans);
+  font-weight: var(--font-weight-normal, 400);
+  font-size: 16px;
```

Comment:
**Issue**: This should remain `1em`. As pointed out, `16px` is the de facto default font size set by browser user agent styles and `1em` will be computed to `16px` with the benefit that changing the default font size in your browser will be reflected by that. Using a non-relative unit will break this.

Reply:
Same reasoning as above/previous comment. Notice this doing this solves the majority of our weird font-sizings issues. to be fair, if anything this should be on `html` but I'd rather stick to the `pseudo-standard` of other projects where they use `body`

Reply:
We shouldn’t sacrifice this feature just so we can drop `--KBadgeFontSize: 1rem;`.

Reply:
Please see screengrabs for what this fixes for font-sizing. Its also exactly the same as Konnect.

Reply:
Whatever this change fixes needs to be accomplished in another way because this introduces the bug of not being able to change the browser font size.

Reply:
Why does it needs to be accomplished in another way, this code here is exactly the same as Konnect.

> introduces the bug of not being able to change the browser font size.

Could you explain a little more. I can still change my browser font size, maybe I'm misunderstanding what you mean?

Reply:
> Why does it needs to be accomplished in another way, this code here is exactly the same as Konnect.

Because it introduces a bug and there is nothing wrong with `font-size: 1em` on `body` to begin with.

> Could you explain a little more. I can still change my browser font size, maybe I'm misunderstanding what you mean?

Changing the font size in your browser (e.g. in Chrome from Medium to Large) changes what the browser puts into its user agent styles. As a result, `font-size: 1em` on `body` will compute to whatever that default is. Medium happens to be 16px while Large works out to 20px. And this works in Kuma GUI, but it breaks when using non-relative units.

Reply:
How do you change the font size in your browser?

Reply:
I don’t know which browser you’re using, but usually they have search features nowadays. A query for "font size" should lead you to the right place.

Reply:
Well I was asking how _you_ do it. Do you _ever_ do it yourself?

As mentioned above, I did it how I would usually do it and my font size change perfectly fine, so I don't see what the problem is? Which is why I thought I may have misunderstood 🤷‍♂️ 

![Screenshot 2023-06-01 at 16 06 30](https://github.com/kumahq/kuma-gui/assets/554604/8c778a7a-5e7c-4ffc-b644-c2ad4d439f81)

Reply:
Yeah right, that's kinda why I was asking how _you_ do it, I never use this for example. How many folks do you think use this when "Page Zoom" is available over this font size preference?

Reply:
Also note that most browsers have this in the top level menus, and is even referred to as you can see in my above browser grab as "Make Text Larger"

Reply:
I think the thing that trumps all this is, if this is how it is in Konnect, and somebody somewhere did actually use what you outlined in your screengrabs as opposed to just using the "Page Zoom" to set their font size. Nothing else in Konnect would change, which I would consider a bug also as its inconsistent.

So to sum up here, if you have a larger issue with not being able to use "Page Zoom" maybe you should raise it in Konnect-land. If they change, we will to.

Reply:
For a bit more context, I used my browser preferences instead of "Page Zoom / Make Text Larger" to set the font size (my browser only gives me a very small range of sizes it seems). I then tried out a couple of popular website, namely Google and Github (this very PR). Github doesn't change at all (but does if I use Page Zoom). Google does have some text that changes, but only 1 piece of text, noticed the button text doesn't change at all.

Considering this and even more specifically our tech audience, how likely is it that somebody is using this font-size preference over something like "Page Zoom"?

![Screenshot 2023-06-01 at 16 36 16](https://github.com/kumahq/kuma-gui/assets/554604/d7eba3d5-b358-4070-91e6-e081e4121b1c)

Honestly I don't see me making any changes here, so I think we should either resolve, or raise it with the rest of the team? Lmk

Reply:
Just because other implementations make the same mistake shouldn't mean need to make it, too, _especially_ if we have ways to avoid it. We can easily address a Kongponents component using relative font sizes and becoming too small/big by setting its variables (which is very much the intended way Kongponents favors for controlling theming properties).

I'm not okay with this change because of the bug I mentioned. Also, it's really not necessary.

Reply:
Bit more information here:

Here is our app on `master` (i.e. previous to this PR)

![Screenshot 2023-06-02 at 09 21 06](https://github.com/kumahq/kuma-gui/assets/554604/e99d9a3d-6881-4b24-be2f-831159f9a7db)

That looks pretty broken currently to me if you are using the browser font-size preference to make the text bigger.

Here's the application with this PR applied and using relative units for font sizing if you are using the browser font-size preference to make the text bigger:

![Screenshot 2023-06-02 at 09 23 50](https://github.com/kumahq/kuma-gui/assets/554604/d6c6307d-3f10-4cda-a049-ecb45add84ce)

Note I've only used relative font sizes on the menu items here.

Using a base font of 16px for a font size and then using relative units is pretty standard practice is quite a common almost "pseudo standard" approach in quite a few things I've been involved with including Konnect.

To finish up here:

1. Konnect uses a base font of 16px
2. As far as I'm aware most folks use "Page Zoom" nowadays if they want to increase their text size.
3. Some of the most popular sites on the internet do not increase their font size when using the font size browser preference.
4. This PR fixes a good few issues we have with messy looking font-sizing
5. If you want our application to increase font size for you when you increase the text size using your font size browser preference, then we should use a 16px base size (for the additional reasons as explained above), and then specify our font-size in relative units. As far as I am aware this is the most common approach.

Each one of those above reasons would be enough for me to apply this PR, but when taken al together I can't see I reason why we should not do this.

Let me know what you think.

Reply:
All of this works just as well with using `1em` instead of `16px` as the base font size **and** it doesn’t break the browser font size setting.

It may well be common practice but it’s certainly more problematic. We can do better and should. We shouldn’t assume what browser features people make use of to access a page. Since it’s easy for us to not break this feature, we should just do it.

> If you want our application to increase font size for you when you increase the text size using your font size browser preference, then we should use a 16px base size (for the additional reasons as explained above), and then specify our font-size in relative units. As far as I am aware this is the most common approach.

That’s not the best approach though. You _can totally_ use a relative base font size **and** relative font sizes for everything else and get the same benefit **without** breaking the browser font size setting.

This may be a common approach, but I pointed out how flawed it is.

All the things fixed in this PR can be fixed without breaking this.

Reply:
@lahabana following the chat we've all just had could you confirm the outcome of this conversation for me? The PR is still in "Changes Requested" and I'm unsure if there is anything else to be done here? lmk

Reply:
As per this conversation it was decided to use `font-size: 16px` and so I’m resolving this thread.